tests/sync_regression.rs

```diff
@@ -0,0 +1,53 @@
+extern crate random_access_memory as ram;
```

Comment:
What's the idea of this file? Just a separation of tests into api tests and bug tests?

Reply:
yeah, pretty much!

When a test fails you can re-run it with `cargo test --test <filename>`, which also made testing for these specific regressions locally a lot faster. Also really like writing a postmortem, so it becomes clear which mistakes we've made in the past - as some of these random inputs can get kinda weird over time.

Eventually we should introduce stuff like failpoints into this https://github.com/pingcap/fail-rs, so we can check how writes persist between shutdowns / random crashes. Things might get super weird at that point, so it'll be good to check!